### PR TITLE
♻️Use core/assert within log assert

### PR DESCRIPTION
**Pull request by rcebulko**
Companion to https://github.com/ampproject/amphtml/pull/33434
Use core assertion logic within otherwise-convoluted log.js logic.

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -399,34 +399,14 @@ export class Log {
...
-      e.fromAssert = true;
-      e.associatedElement = firstElement;
-      e.messageArray = messageArray;
-      this.prepareError_(e);
-      // __AMP_REPORT_ERROR is installed globally per window in the entry point.
-      self.__AMP_REPORT_ERROR(e);
-      throw e;
+
+    try {
+      return assertion(this == logs.user ? UserError : Error, shouldBeTrueish);
```
Comment (jridgewell) [inline src/log.js:404]: Couldn't we import `pureUserAssert`/`pureDevAssert` and avoid this `UserError`/`Error` part? Let pure asserts figure out which error it's throwing.

File: src/log.js [hunk 2]
```diff
@@ -399,34 +399,18 @@ export class Log {
...
-      // __AMP_REPORT_ERROR is installed globally per window in the entry point.
-      self.__AMP_REPORT_ERROR(e);
-      throw e;
+
+    try {
+      const assertion = this == logs.user ? pureUserAssert : pureDevAssert;
+      return assertion.apply(
+        null,
+        [shouldBeTrueish, opt_message].concat(var_args)
+      );
```
Comment (jridgewell) [inline src/log.js:408]:
```suggestion
      return assertion.apply(null, arguments);
```

File: src/core/assert.js [hunk 3]
```diff
@@ -91,6 +91,7 @@ function assertion(errorCls, shouldBeTruthy, opt_message, var_args) {
   });
 
   const error = new errorCls(message);
+  error.fromAssert = true;
```
Comment (jridgewell) [inline src/core/assert.js:94]: Do we need this?

**Review by jridgewell — APPROVED** [review]

File: src/log.js (hunk 2 above)
Comment (rcebulko) [reply src/log.js:408]: *facepalm*

File: src/core/assert.js (hunk 3 above)
Comment (rcebulko) [reply src/core/assert.js:94]: I have no idea; trying to make things work and understand why they don't/what depends on what. This PR should probably be in draft for now, sorry.

File: src/log.js (hunk 1 above)
Comment (rcebulko) [reply src/log.js]: Done.

**Comment by rcebulko** [comment]
I'm not sure why this is inflating the bundle-size, digging through binaries to try and compare

**Comment by rcebulko** [comment]
@jridgewell This increases a lot of bundle-sizes slightly, but I'm not sure how to avoid it (until more migration is done) since there's an extra layer or two going through AMP and core assertion helpers. v0.js increases, v0.mjs decreases slightly

**Review by jridgewell — APPROVED** [review]

**Pull request merged**